src/parsec_service/tests/existing_tests.md

```diff
@@ -0,0 +1,55 @@
+|-----------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [TPM integration tests](https://github.com/parallaxsecond/parsec/tree/master/tests/providers) | These tests initialize a provider structure and perform operation using the `Provide` trait. A software crypto library, like `ring`, can be used to verify the behaviour. [Example with the TPM provider](https://github.com/parallaxsecond/parsec/blob/master/tests/providers/tpm.rs). |
+
+## End-to-end testing
+
+These tests are meant to be executed on a running Parsec service and using a Parsec client to
+communicate to it. The
+[`parsec/e2e_tests`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests) crate contains
+a test client based on the Rust client that is used inside the tests written in
+[`parsec/e2e_tests/tests`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests).
+
+The end-to-end tests contain tests to be executed: * with a Parsec service containing a single
+provider:
+[`per_provider`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/per_provider)
+tests * with a Parsec service containing all providers:
+[`all_providers`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/all_providers)
+and [`config`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/config) tests
+
+The
+[`parsec/e2e_tests/provider_cfg`](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/provider_cfg)
+folder contain the `Dockerfile` and Parsec configuration needed to run Parsec for the corresponding
+tests.
+
+| Name                                                                                                                                     | Description                                                                                                       |
+|------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
+| [All providers e2e tests](https://github.com/parallaxsecond/parsec/tree/master/e2e_tests/tests/all_providers)                            | End-to-end tests on a service with all providers. Mostly to test core operations.                                 |
```

Comment:
You could also mention that now we do "cross testing" - either in this line or in another one, where we test that the output of various operations from different providers is accepted by the other providers.